src/globals/js/watch.js

```diff
@@ -0,0 +1,66 @@
+import initCheckbox from '../../components/checkbox/checkbox';
+import { componentClasses } from '../../index';
+
+const forEach = Array.prototype.forEach;
+
+/**
+ * Automatically instantiates/destroys components in the given element, by watching for DOM additions/removals.
+ * @param {Node} target The DOM node to instantiate components in. Should be a document or an element.
+ * @param {Object} [options] The component options.
+ * @returns {Handle} The handle to stop watching.
+ */
+export default function (target = document, options = {}) {
+  if (target.nodeType !== Node.ELEMENT_NODE && target.nodeType !== Node.DOCUMENT_NODE) {
+    throw new TypeError('DOM document or DOM element should be given to watch for DOM node to create/release components.');
+  }
+
+  const handles = componentClasses.map(Clz => Clz.init(target, options)).filter(Boolean);
+  handles.push(initCheckbox());
+
+  const componentClassesForWatchInit = componentClasses.filter(Clz => !Clz.forLazyInit);
+  let observer = new MutationObserver((records) => {
+    records.forEach((record) => {
+      forEach.call(record.addedNodes, (node) => {
```

Comment:
this function gets fairly long - for readability, could you refactor the forEach loops into separate functions?